api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/LocaleUtility.java

```diff
@@ -10,10 +10,14 @@ private LocaleUtility(){}
 
     public static Locale languageStringToLocale(String localeString){
         String[] parsedLoc = localeString.trim().split("_", -1);
-        //regex pattern for locale tag with script specified
-        Locale locale = localeString.matches("^[a-z]{1,3}_[A-Z][a-z]{3}_[A-Z]{2}") ?
-            new Locale.Builder().setLanguage(parsedLoc[0]).setRegion(parsedLoc[2]).setScript(parsedLoc[1]).build() :
-            LocaleUtils.toLocale(localeString);
+        Locale locale =
```

Comment:
The previous version of the code already seems like an abuse of the ternary operator but this version nests an additional layer.  Can this be made easier to read with regular if/else blocks?